Tasks.md

```diff
@@ -1,94 +1,301 @@
+        <div class="mole"></div>
+    </div>
+    <div class="hole">
+        <div class="mole"></div>
+    </div>
+    <div class="hole">
+        <div class="mole"></div>
+    </div>
+    <div class="hole">
+        <div class="mole"></div>
+    </div>
+    <div class="hole">
+        <div class="mole"></div>
+    </div>
+    <div class="hole">
+        <div class="mole"></div>
+    </div>
+</div>
+```
+
+*Javascript*
+```javascript
+// ...
+
+for (let mole of moles) {
+    mole.addEventListener('click', whack)
+}
+```
+</details>
+
+
+### 7) RANDOM MOLE
+- above `startGame()`, write a function that returns a random mole (i.e.: a random element from the moles array)
+    - think about a good name for this function
+    - first select a random index: `Math.floor(Math.random() * moles.length)`
+    - then access that index of the `moles` array
+    - return that element
+- inside `peep()`, call the function you created and assign it to the `mole` variable
+
+<details>
+<summary>Feeling lost? Expand to get a hint!</summary>
+
+```javascript
+// returns a random element from the holes array
+function randomMole() {
+    const index = Math.floor(Math.random() * moles.length)
+    const mole = moles[index]
+    return mole
+}
+
+async function peep() {
+    const mole = randomMole()
+    // ...
+}
+```
+</details>
+
+
+### 8) SCORE
+- in the beginning of the file, add a variable for the score
+    - question: do we use `const` or `let`?
+- set it to `0` when the game starts, before the call to `peep()`
+- show it when the game starts (update the HTML `scoreBoard` element, use the `textContent` property)
+- increase it by 1 when you wack a mole (i.e. in the `whack()` function)
+    - hint: first, write the expression for "your score plus 1 point". Then assign this expression to the score-keeping variable
+- show the new score immediately, like we did in `startGame()`
+
+<details>
+<summary>Feeling lost? Expand to get a hint!</summary>
+
+```javascript
+let score
+
+// ...
+
+function startGame() {
+    score = 0
+    scoreBoard.textContent = score
+    // ...
+}
+
+function whack(event) {
+    // ...
+    score = score + 1
+    scoreBoard.textContent = score
+}
+```
+</details>
+
+
+### 9) TIMEUP
+- add a variable for whether the time is up or not
+    - which type?
+- inside `peep()`, add the condition that you don't do anything if the time is up!
+- when the game starts, set the variable to `false`
+- after 15 seconds, finish the game by setting the variable to `true`
+    - hint: create a function `finish()` which sets the variable (to what?)
+    - before calling `finish()`, use the `sleep(time)` function
+    - before `sleep` write `await` and make the `startGame()` function `async`
+
+<details>
+<summary>Feeling lost? Expand to get a hint!</summary>
+
+```javascript
+let timeUp
+
+// ...
+
+function finish() {
+    timeUp = true
+}
+
+async function startGame() {
+    timeUp = false
+    // ...
+    await sleep(15000)
+    finish()
+}
```

Comment:
Same thing here: maybe we can add those 3 lines all below `// ...` and then also change `09.js` accordingly

Reply:
Would love to do that but `timeUp = false` has to go before `peep()` because peep will do nothing otherwise:
```javascript
async function peep() {
    if (!timeUp) {
        // ...
    }
}
```

Reply:
ah of course 😅